### PR TITLE
[full-ci] Add per-user storage view

### PR DESCRIPTION
**Pull request by pako81**
## Description
Add per-user storage view

## Related Issue
- https://github.com/owncloud/enterprise/issues/5468

## Motivation and Context
Make possible to visualise used storage on per-user basis

## How Has This Been Tested?
- Access ~/settings/users URL. A new column `Storage Used` will display the amount of used storage for every user. This option can be also opted in/out under `Settings` in order to align with all other available users' options.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [X] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised
- [X] Changelog item

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISettingsMenu-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/38026/133

File: changelog/unreleased/40657 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Add per-user storage view
+
+Adding option for displaying used storage on per-user basis
```
Comment (jnweiger) [inline changelog/unreleased/40657:3]:
```suggestion
Adding option for displaying used storage on per-user basis.
```

**Comment by phil-davis** [comment]
```
$ make test-php-unit NOCOVERAGE=true TEST_PHP_SUITE=tests/Settings/Controller/UsersControllerTest.php 
PHPUNIT="/home/phil/git/owncloud/core/lib/composer/phpunit/phpunit/phpunit" build/autotest.sh sqlite tests/Settings/Controller/UsersControllerTest.php
Using PHP executable /usr/bin/php
Parsing all files in lib/public for the presence of @since or @deprecated on each method...

Using database oc_autotest
Setup environment for sqlite testing on local storage ...
Updated 0 paths from the index
Installing ....
creating sqlite db
ownCloud was successfully installed
Testing with sqlite ...
No coverage
/home/phil/git/owncloud/core/lib/composer/phpunit/phpunit/phpunit --configuration phpunit-autotest.xml --log-junit autotest-results-sqlite.xml ../tests/Settings/Controller/UsersControllerTest.php 
PHPUnit 9.6.3 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.33
Configuration: phpunit-autotest.xml

EEEE.EEEE..........EEEEEE.............................E.......... 65 / 73 ( 89%)
.......E                                                          73 / 73 (100%)

Time: 00:00.403, Memory: 32.00 MB

There were 16 errors:

1) Tests\Settings\Controller\UsersControllerTest::testIndexAdmin
Error: Call to a member function getFileInfo() on null

/home/phil/git/owncloud/core/lib/private/Files/Filesystem.php:1030
/home/phil/git/owncloud/core/lib/private/legacy/helper.php:582
/home/phil/git/owncloud/core/settings/Controller/UsersController.php:233
/home/phil/git/owncloud/core/settings/Controller/UsersController.php:339
/home/phil/git/owncloud/core/tests/Settings/Controller/UsersControllerTest.php:354
```

Unit tests are failing in CI and locally for me. Is the code different from what was in the previous PR #40506 ?

**Comment by pako81** [comment]
The code should be exactly the same and it proves to work locally for me.

**Comment by phil-davis** [comment]
> The code should be exactly the same and it proves to work locally for me.

"somebody" needs to investigate why those unit tests are failing, and fix them.

File: changelog/unreleased/40657 [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Add per-user storage view
+
+Adding option for displaying used storage on per-user basis.
```
Comment (jnweiger) [inline changelog/unreleased/40657:3]:
```suggestion
Adding an option for displaying used storage on per-user basis.
```

**Comment by prsnbrg** [comment]
From my point of view, the used space is always given in MB. This is not user friendly in my view for over 100 GB.

A possible solution can be `\OC_Helper::humanFileSize()`

**Comment by pako81** [comment]
> From my point of view, the used space is always given in MB. This is not user friendly in my view for over 100 GB.

Thanks for the pointer - indeed it seems better to use `humanFileSize()`. Changed it accordingly.

**Comment by pako81** [comment]
I have adjusted the string for not-computed storage to 0 B. This is consistent with the case where user has no files at all --> in the personal settings page is currently reported `You are using 0 B`

Additionally, I have increased the log level to `debug` otherwise an entry at level 3 would be logged for every user who has not logged in yet.

**Comment by pako81** [comment]
@jvillafanez do you see something in the UsersController code which could explain the 16 failing tests?

```
1) Tests\Settings\Controller\UsersControllerTest::testIndexAdmin
Error: Call to a member function getFileInfo() on null

/home/phil/git/owncloud/core/lib/private/Files/Filesystem.php:1030
/home/phil/git/owncloud/core/lib/private/legacy/helper.php:582
/home/phil/git/owncloud/core/settings/Controller/UsersController.php:233
/home/phil/git/owncloud/core/settings/Controller/UsersController.php:339
/home/phil/git/owncloud/core/tests/Settings/Controller/UsersControllerTest.php:354

**Comment by jvillafanez** [comment]
The filesystem is likely mocked or not initialized in the tests.
I guess the problem is that the `\OC_Helper::getStorageInfo('/');` call can't be mocked. We might need a way to mock that call somehow so it doesn't go all the way through.

**Comment by mrow4a** [comment]
@pako81 isnt it better to use Metrics App for this? 

```
cd /var/www/owncloud/apps/metrics
make build-js
make vendor
composer install

occ a:e metrics
occ config:system:set "metrics_shared_secret" --value 1234
```

Screenshot 2023-03-10 at 15 46 46

**Comment by mrow4a** [comment]
@pako81 take decision and close if so

**Comment by jnweiger** [comment]
@mrow4a is right, this information is more logically found in the metrics app.
But as we have quota in the users listing, and each user sees his own storage usage + his quota, I would not mind to give the admin an overview already in the users listing.  The column is optional, so it does not take away screen space for thse who don't want to have it.

**Pull request closed without merging**